flutter/lib/src/screenshot/screenshot_attachment.dart

```diff
@@ -0,0 +1,63 @@
+import 'dart:async';
+import 'dart:typed_data';
+import 'dart:ui' as ui show ImageByteFormat;
+
+import 'package:flutter/rendering.dart';
+import 'package:flutter/scheduler.dart';
+import 'package:sentry/sentry.dart';
+
+import '../sentry_widget.dart';
+
+class ScreenshotAttachment implements SentryAttachment {
+  final SchedulerBinding _schedulerBinding;
+  final SentryOptions _options;
+
+  ScreenshotAttachment(this._schedulerBinding, this._options);
+
+  @override
+  String attachmentType = SentryAttachment.typeAttachmentDefault;
+
+  @override
+  String? contentType = 'image/png';
+
+  @override
+  String filename = 'screenshot.png';
+
+  @override
+  bool addToTransactions = true;
+
+  @override
+  FutureOr<Uint8List> get bytes async {
+    final _completer = Completer<Uint8List?>();
+    // We add an post frame callback because we aren't able to take a screenshot
+    // if there's currently a draw in process.
+    _schedulerBinding.addPostFrameCallback((timeStamp) async {
+      final screenshot = await _createScreenshot();
+      _completer.complete(screenshot);
+    });
+    return await _completer.future ?? Uint8List.fromList([]);
+  }
+
+  Future<Uint8List?> _createScreenshot() async {
```

Comment:
Maybe this should be an `EventProcessor`, so when events are caught, if this processor is there, it takes the screenshot.

Reply:
The approach is taken from the cocoa SDK. They event went further and support an array of attachment processors, and not just one. https://github.com/getsentry/sentry-cocoa/blob/58f558dc58731af7d6522987fe5e95b0440025a0/Sources/Sentry/SentryClient.m#L356